From 78fd45a343db3b0866df7f04381063d04e6c8c0e Mon Sep 17 00:00:00 2001
From: Amrita Raju <amrita.raju@intel.com>
Date: Fri, 8 Feb 2019 09:55:05 +0530
Subject: [PATCH] wifi: Disable capabilities not yet supported

As E2E support is not yet implemented for following capabilities,
remove the capabilities.
DEBUG_RING_BUFFER_VENDOR_DATA, DEBUG_HOST_WAKE_REASON_STATS,
DEBUG_ERROR_ALERTS and APF.

Change-Id: I3c33a9ad826e36f3765bb57b11e50b9ad92c1221
Tracked-On: OAM-75990
Signed-off-by: Amrita Raju <amrita.raju@intel.com>
Signed-off-by: Jeevaka Prabu Badrappan <jeevaka.badrappan@intel.com>
---
 wifi/1.2/default/hidl_struct_util.cpp | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/wifi/1.2/default/hidl_struct_util.cpp b/wifi/1.2/default/hidl_struct_util.cpp
index 39ac544e6..5555d6a7b 100644
--- a/wifi/1.2/default/hidl_struct_util.cpp
+++ b/wifi/1.2/default/hidl_struct_util.cpp
@@ -128,7 +128,7 @@ bool convertLegacyFeaturesToHidlChipCapabilities(
         return false;
     }
     *hidl_caps = {};
-    using HidlChipCaps = IWifiChip::ChipCapabilityMask;
+    //using HidlChipCaps = IWifiChip::ChipCapabilityMask;
     for (const auto feature : {legacy_hal::WIFI_LOGGER_MEMORY_DUMP_SUPPORTED,
                                legacy_hal::WIFI_LOGGER_DRIVER_DUMP_SUPPORTED,
                                legacy_hal::WIFI_LOGGER_CONNECT_EVENT_SUPPORTED,
@@ -148,9 +148,9 @@ bool convertLegacyFeaturesToHidlChipCapabilities(
     }
     // There are no flags for these 3 in the legacy feature set. Adding them to
     // the set because all the current devices support it.
-    *hidl_caps |= HidlChipCaps::DEBUG_RING_BUFFER_VENDOR_DATA;
-    *hidl_caps |= HidlChipCaps::DEBUG_HOST_WAKE_REASON_STATS;
-    *hidl_caps |= HidlChipCaps::DEBUG_ERROR_ALERTS;
+    //*hidl_caps |= HidlChipCaps::DEBUG_RING_BUFFER_VENDOR_DATA;
+    //*hidl_caps |= HidlChipCaps::DEBUG_HOST_WAKE_REASON_STATS;
+    //*hidl_caps |= HidlChipCaps::DEBUG_ERROR_ALERTS;
     return true;
 }
 
@@ -350,7 +350,7 @@ bool convertLegacyFeaturesToHidlStaCapabilities(
         return false;
     }
     *hidl_caps = {};
-    using HidlStaIfaceCaps = IWifiStaIface::StaIfaceCapabilityMask;
+    //using HidlStaIfaceCaps = IWifiStaIface::StaIfaceCapabilityMask;
     for (const auto feature : {legacy_hal::WIFI_LOGGER_PACKET_FATE_SUPPORTED}) {
         if (feature & legacy_logger_feature_set) {
             *hidl_caps |=
@@ -370,7 +370,7 @@ bool convertLegacyFeaturesToHidlStaCapabilities(
     }
     // There is no flag for this one in the legacy feature set. Adding it to the
     // set because all the current devices support it.
-    *hidl_caps |= HidlStaIfaceCaps::APF;
+    //*hidl_caps |= HidlStaIfaceCaps::APF;
     return true;
 }
 
-- 
2.17.1

